tdrs-backend/setup.cfg

```diff
@@ -1,26 +1,46 @@
-[flake8]
-# D203 conflicts with D211, which is the more conventional of the two
-ignore = E226,E302,E41,D203
-max-line-length = 160
-max-complexity = 10
+[coverage:run]
+branch = True
+omit =
+    tdpservice/settings/development.py
+    tdpservice/settings/local.py
+    tdpservice/settings/production.py
+    tdpservice/settings/staging.py
+    tdpservice/wsgi.py
+
+[coverage:report]
+ignore_errors = True
+omit =
+    tdpservice/settings/development.py
+    tdpservice/settings/local.py
+    tdpservice/settings/production.py
+    tdpservice/settings/staging.py
+    tdpservice/wsgi.py
 
+[flake8]
 docstring-convention=numpy
 exclude =
-    # No need to traverse our git directory
     .git,
-    # There's no value in checking cache directories
     __pycache__,
-    # The conf file is mostly autogenerated, ignore it
-    docs/source/conf.py,
-    # The old directory contains Flake8 2.0
-    old,
-    # This contains our built documentation
-    build,
-    # This contains builds of flake8 that we don't want to check
-    dist,
-    # No need to check migrations
-    migrations,
-    # No need to check blank files
-    __init__.py,
-    # No need to check tests for linting
-    tests/*
+    # We ideally want migrations to be readbale, but they often get quite nested and
+    # often have long strings. So we auto-format with black but ignore them here.
+    */migrations/*,
+extend-ignore = D104, D106, D203, E203, W503
```

Comment:
We need to document what these codes are and why we need to omit them. They will flag this.